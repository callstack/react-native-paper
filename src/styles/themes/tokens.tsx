import { Platform } from 'react-native';

import type { Font } from '../../types';

const ref = {
  palette: {
    primary100: 'rgba(255, 255, 255, 1)',
    primary99: 'rgba(255, 251, 254, 1)',
    primary95: 'rgba(246, 237, 255, 1)',
    primary90: 'rgba(234, 221, 255, 1)',
    primary80: 'rgba(208, 188, 255, 1)',
    primary70: 'rgba(182, 157, 248, 1)',
    primary60: 'rgba(154, 130, 219, 1)',
    primary50: 'rgba(127, 103, 190, 1)',
    primary40: 'rgba(103, 80, 164, 1)',
    primary30: 'rgba(79, 55, 139, 1)',
    primary20: 'rgba(56, 30, 114, 1)',
    primary10: 'rgba(33, 0, 93, 1)',
    primary0: 'rgba(0, 0, 0, 1)',
    secondary100: 'rgba(255, 255, 255, 1)',
    secondary99: 'rgba(255, 251, 254, 1)',
    secondary95: 'rgba(246, 237, 255, 1)',
    secondary90: 'rgba(232, 222, 248, 1)',
    secondary80: 'rgba(204, 194, 220, 1)',
    secondary70: 'rgba(176, 167, 192, 1)',
    secondary60: 'rgba(149, 141, 165, 1)',
    secondary50: 'rgba(122, 114, 137, 1)',
    secondary40: 'rgba(98, 91, 113, 1)',
    secondary30: 'rgba(74, 68, 88, 1)',
    secondary20: 'rgba(51, 45, 65, 1)',
    secondary10: 'rgba(29, 25, 43, 1)',
    secondary0: 'rgba(0, 0, 0, 1)',
    tertiary100: 'rgba(255, 255, 255, 1)',
    tertiary99: 'rgba(255, 251, 250, 1)',
    tertiary95: 'rgba(255, 236, 241, 1)',
    tertiary90: 'rgba(255, 216, 228, 1)',
    tertiary80: 'rgba(239, 184, 200, 1)',
    tertiary70: 'rgba(210, 157, 172, 1)',
    tertiary60: 'rgba(181, 131, 146, 1)',
    tertiary50: 'rgba(152, 105, 119, 1)',
    tertiary40: 'rgba(125, 82, 96, 1)',
    tertiary30: 'rgba(99, 59, 72, 1)',
    tertiary20: 'rgba(73, 37, 50, 1)',
    tertiary10: 'rgba(49, 17, 29, 1)',
    tertiary0: 'rgba(0, 0, 0, 1)',
    neutral100: 'rgba(255, 255, 255, 1)',
    neutral99: 'rgba(255, 251, 254, 1)',
    neutral95: 'rgba(244, 239, 244, 1)',
    neutral90: 'rgba(230, 225, 229, 1)',
    neutral80: 'rgba(201, 197, 202, 1)',
    neutral70: 'rgba(174, 170, 174, 1)',
    neutral60: 'rgba(147, 144, 148, 1)',
    neutral50: 'rgba(120, 117, 121, 1)',
    neutral40: 'rgba(96, 93, 98, 1)',
    neutral30: 'rgba(72, 70, 73, 1)',
    neutral20: 'rgba(49, 48, 51, 1)',
    neutral10: 'rgba(28, 27, 31, 1)',
    neutral0: 'rgba(0, 0, 0, 1)',
    neutralVariant100: 'rgba(255, 255, 255, 1)',
    neutralVariant99: 'rgba(255, 251, 254, 1)',
    neutralVariant95: 'rgba(245, 238, 250, 1)',
    neutralVariant90: 'rgba(231, 224, 236, 1)',
    neutralVariant80: 'rgba(202, 196, 208, 1)',
    neutralVariant70: 'rgba(174, 169, 180, 1)',
    neutralVariant60: 'rgba(147, 143, 153, 1)',
    neutralVariant50: 'rgba(121, 116, 126, 1)',
    neutralVariant40: 'rgba(96, 93, 102, 1)',
    neutralVariant30: 'rgba(73, 69, 79, 1)',
    neutralVariant20: 'rgba(50, 47, 55, 1)',
    neutralVariant10: 'rgba(29, 26, 34, 1)',
    neutralVariant0: 'rgba(0, 0, 0, 1)',
    error100: 'rgba(255, 255, 255, 1)',
    error99: 'rgba(255, 251, 249, 1)',
    error95: 'rgba(252, 238, 238, 1)',
    error90: 'rgba(249, 222, 220, 1)',
    error80: 'rgba(242, 184, 181, 1)',
    error70: 'rgba(236, 146, 142, 1)',
    error60: 'rgba(228, 105, 98, 1)',
    error50: 'rgba(220, 54, 46, 1)',
    error40: 'rgba(179, 38, 30, 1)',
    error30: 'rgba(140, 29, 24, 1)',
    error20: 'rgba(96, 20, 16, 1)',
    error10: 'rgba(65, 14, 11, 1)',
    error0: 'rgba(0, 0, 0, 1)',
  },

  typeface: {
    brandRegular: Platform.select({
      web: 'Roboto, "Helvetica Neue", Helvetica, Arial, sans-serif',
      ios: 'System',
      default: 'sans-serif',
    }),
    weightRegular: '400' as Font['fontWeight'],

    plainMedium: Platform.select({
      web: 'Roboto, "Helvetica Neue", Helvetica, Arial, sans-serif',
      ios: 'System',
      default: 'sans-serif-medium',
    }),
    weightMedium: '500' as Font['fontWeight'],
  },

  opacity: {
    level1: 0.08,
    level2: 0.12,
    level3: 0.16,
    level4: 0.38,
  },
};

const regularType = {
  fontFamily: ref.typeface.brandRegular,
  letterSpacing: 0,
  fontWeight: ref.typeface.weightRegular,
};

const mediumType = {
  fontFamily: ref.typeface.plainMedium,
  letterSpacing: 0.15,
  fontWeight: ref.typeface.weightMedium,
};

export const typescale = {
  displayLarge: {
    ...regularType,
    lineHeight: 64,
    fontSize: 57,
  },
  displayMedium: {
    ...regularType,
    lineHeight: 52,
    fontSize: 45,
  },
  displaySmall: {
    ...regularType,
    lineHeight: 44,
    fontSize: 36,
  },

  headlineLarge: {
    ...regularType,
    lineHeight: 40,
    fontSize: 32,
  },
  headlineMedium: {
    ...regularType,
    lineHeight: 36,
    fontSize: 28,
  },
  headlineSmall: {
    ...regularType,
    lineHeight: 32,
    fontSize: 24,
  },

  titleLarge: {
    ...regularType,
    lineHeight: 28,
    fontSize: 22,
  },
  titleMedium: {
    ...mediumType,
    lineHeight: 24,
    fontSize: 16,
  },
  titleSmall: {
    ...mediumType,
    letterSpacing: 0.1,
    lineHeight: 20,
    fontSize: 14,
  },

  labelLarge: {
    ...mediumType,
    letterSpacing: 0.1,
    lineHeight: 20,
    fontSize: 14,
  },
  labelMedium: {
    ...mediumType,
    letterSpacing: 0.5,
    lineHeight: 16,
    fontSize: 12,
  },
  labelSmall: {
    ...mediumType,
    letterSpacing: 0.5,
    lineHeight: 16,
    fontSize: 11,
  },

  bodyLarge: {
    ...mediumType,
    fontWeight: ref.typeface.weightRegular,
    fontFamily: ref.typeface.brandRegular,
    lineHeight: 24,
    fontSize: 16,
  },
  bodyMedium: {
    ...mediumType,
    fontWeight: ref.typeface.weightRegular,
    fontFamily: ref.typeface.brandRegular,
    letterSpacing: 0.25,
    lineHeight: 20,
    fontSize: 14,
  },
  bodySmall: {
    ...mediumType,
    fontWeight: ref.typeface.weightRegular,
    fontFamily: ref.typeface.brandRegular,
    letterSpacing: 0.4,
    lineHeight: 16,
    fontSize: 12,
  },

  default: {
    ...regularType,
  },
};

export const tokens = {
  md: {
    ref,
    sys: {
      typescale,
    },
  },
};

export const Colors = ref.palette;
