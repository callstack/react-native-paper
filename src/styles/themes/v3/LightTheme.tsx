import { I18nManager } from 'react-native';

import color from 'color';

import { MD3Colors, tokens } from './tokens';
import type { MD3Theme } from '../../../types';
import configureFonts from '../../fonts';

const { palette, opacity } = tokens.md.ref;

export const MD3LightTheme: MD3Theme = {
  dark: false,
  direction: I18nManager.getConstants().isRTL ? 'rtl' : 'ltr',
  roundness: 4,
  version: 3,
  isV3: true,
  colors: {
    primary: palette.primary40,
    primaryContainer: palette.primary90,
    secondary: palette.secondary40,
    secondaryContainer: palette.secondary90,
    tertiary: palette.tertiary40,
    tertiaryContainer: palette.tertiary90,
    surface: palette.neutral99,
    surfaceVariant: palette.neutralVariant90,
    surfaceDisabled: color(palette.neutral10)
      .alpha(opacity.level2)
      .rgb()
      .string(),
    background: palette.neutral99,
    error: palette.error40,
    errorContainer: palette.error90,
    onPrimary: palette.primary100,
    onPrimaryContainer: palette.primary10,
    onSecondary: palette.secondary100,
    onSecondaryContainer: palette.secondary10,
    onTertiary: palette.tertiary100,
    onTertiaryContainer: palette.tertiary10,
    onSurface: palette.neutral10,
    onSurfaceVariant: palette.neutralVariant30,
    onSurfaceDisabled: color(palette.neutral10)
      .alpha(opacity.level4)
      .rgb()
      .string(),
    onError: palette.error100,
    onErrorContainer: palette.error10,
    onBackground: palette.neutral10,
    outline: palette.neutralVariant50,
    outlineVariant: palette.neutralVariant80,
    inverseSurface: palette.neutral20,
    inverseOnSurface: palette.neutral95,
    inversePrimary: palette.primary80,
    shadow: palette.neutral0,
    scrim: palette.neutral0,
    backdrop: color(MD3Colors.neutralVariant20).alpha(0.4).rgb().string(),
    elevation: {
      level0: 'transparent',
      // Note: Color values with transparency cause RN to transfer shadows to children nodes
      // instead of View component in Surface. Providing solid background fixes the issue.
      // Opaque color values generated with `palette.primary99` used as background
      level1: 'rgb(247, 243, 249)', // palette.primary40, alpha 0.05
      level2: 'rgb(243, 237, 246)', // palette.primary40, alpha 0.08
      level3: 'rgb(238, 232, 244)', // palette.primary40, alpha 0.11
      level4: 'rgb(236, 230, 243)', // palette.primary40, alpha 0.12
      level5: 'rgb(233, 227, 241)', // palette.primary40, alpha 0.14
    },
  },
  fonts: configureFonts(),
  animation: {
    scale: 1.0,
  },
};
