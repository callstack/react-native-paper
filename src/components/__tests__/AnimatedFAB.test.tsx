/// <reference types="@testing-library/jest-native" />

import * as React from 'react';
import { Animated, StyleSheet } from 'react-native';

import { fireEvent, render } from '@testing-library/react-native';
import { act } from 'react-test-renderer';

import { MD3Colors } from '../../styles/themes/v3/tokens';
import AnimatedFAB from '../FAB/AnimatedFAB';

const styles = StyleSheet.create({
  background: {
    backgroundColor: MD3Colors.tertiary50,
  },
});

it('renders animated fab', () => {
  const tree = render(
    <AnimatedFAB onPress={() => {}} label="" extended={false} icon="plus" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders animated fab with label on the right by default', () => {
  const tree = render(
    <AnimatedFAB label="text" extended onPress={() => {}} icon="plus" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders animated fab with label on the left', () => {
  const tree = render(
    <AnimatedFAB
      label="text"
      extended
      animateFrom="left"
      onPress={() => {}}
      icon="plus"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders animated fab with only transparent container', () => {
  const { getByTestId } = render(
    <AnimatedFAB
      label="text"
      animateFrom="left"
      onPress={() => {}}
      icon="plus"
      testID="animated-fab"
      extended={false}
      style={styles.background}
    />
  );
  expect(getByTestId('animated-fab-container')).toHaveStyle({
    backgroundColor: 'transparent',
  });
});

it('animated value changes correctly', () => {
  const value = new Animated.Value(1);
  const { getByTestId } = render(
    <AnimatedFAB
      label="text"
      icon="plus"
      testID="animated-fab"
      extended={false}
      style={[{ transform: [{ scale: value }] }]}
    />
  );
  expect(getByTestId('animated-fab-container-outer-layer')).toHaveStyle({
    transform: [{ scale: 1 }],
  });

  Animated.timing(value, {
    toValue: 1.5,
    useNativeDriver: false,
    duration: 200,
  }).start();

  act(() => {
    jest.advanceTimersByTime(200);
  });

  expect(getByTestId('animated-fab-container-outer-layer')).toHaveStyle({
    transform: [{ scale: 1.5 }],
  });
});

it('renders FAB without uppercase styling if uppercase prop is falsy', () => {
  const { getByTestId } = render(
    <AnimatedFAB
      label="text"
      animateFrom="left"
      onPress={() => {}}
      icon="plus"
      testID="animated-fab"
      style={styles.background}
      extended={false}
      uppercase={false}
    />
  );

  expect(getByTestId('animated-fab-text')).not.toHaveStyle({
    textTransform: 'uppercase',
  });
});

it('renders FAB with uppercase styling if uppercase prop is truthy', () => {
  const { getByTestId } = render(
    <AnimatedFAB
      label="text"
      animateFrom="left"
      onPress={() => {}}
      icon="plus"
      testID="animated-fab"
      style={styles.background}
      extended={false}
      uppercase
    />
  );

  expect(getByTestId('animated-fab-text')).toHaveStyle({
    textTransform: 'uppercase',
  });
});

it('renders correct elevation value for shadow views', () => {
  const { getByTestId } = render(
    <AnimatedFAB
      extended
      label="text"
      animateFrom="left"
      onPress={() => {}}
      icon="plus"
      testID="animated-fab"
    />
  );

  expect(getByTestId('animated-fab-shadow')).toHaveStyle({ elevation: 3 });
  expect(getByTestId('animated-fab-extended-shadow')).toHaveStyle({
    elevation: 3,
  });
});

describe('AnimatedFAB events', () => {
  it('onPress passes event', () => {
    const onPress = jest.fn();
    const { getByTestId } = render(
      <AnimatedFAB extended icon="plus" onPress={onPress} label="Add items" />
    );

    act(() => {
      fireEvent(getByTestId('animated-fab'), 'onPress', { key: 'value' });
    });

    expect(onPress).toHaveBeenCalledWith({ key: 'value' });
  });

  it('onLongPress passes event', () => {
    const onLongPress = jest.fn();
    const { getByTestId } = render(
      <AnimatedFAB
        extended
        icon="plus"
        onLongPress={onLongPress}
        label="Add items"
      />
    );

    act(() => {
      fireEvent(getByTestId('animated-fab'), 'onLongPress', { key: 'value' });
    });

    expect(onLongPress).toHaveBeenCalledWith({ key: 'value' });
  });
});
