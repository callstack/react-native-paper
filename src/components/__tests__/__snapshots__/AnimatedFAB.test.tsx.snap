// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders animated fab 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "transparent",
      "borderRadius": 16,
      "opacity": 1,
      "position": "absolute",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="animated-fab-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 16,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    testID="animated-fab-container"
  >
    <View
      collapsable={false}
      style={
        [
          false,
          {
            "height": 56,
          },
          {
            "borderRadius": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "elevation": 0,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 3,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 72,
            }
          }
          testID="animated-fab-extended-shadow"
        />
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 3,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 16,
                },
              ],
              "width": 56,
            }
          }
          testID="animated-fab-shadow"
        />
      </View>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          [
            {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 16,
              "height": 56,
              "left": 16,
              "right": undefined,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 72,
            }
          }
        >
          <View
            accessibilityLabel=""
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 16,
                },
              ]
            }
            testID="animated-fab"
          >
            <View
              style={
                [
                  {
                    "height": 56,
                  },
                  {
                    "borderRadius": 16,
                    "width": 72,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "height": 56,
          "justifyContent": "center",
          "left": 16,
          "position": "absolute",
          "right": undefined,
          "transform": [
            {
              "translateX": 0,
            },
          ],
          "width": 56,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          [
            {
              "color": "rgba(33, 0, 93, 1)",
              "fontSize": 24,
            },
            [
              {
                "lineHeight": 24,
                "transform": [
                  {
                    "scaleX": 1,
                  },
                ],
              },
              {
                "backgroundColor": "transparent",
              },
            ],
          ]
        }
      >
        plus
      </Text>
    </View>
    <View
      pointerEvents="none"
    >
      <Text
        collapsable={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        onTextLayout={[Function]}
        style={
          {
            "color": "rgba(33, 0, 93, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 20,
            "minWidth": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 16,
            "textAlign": "left",
            "top": -28,
            "transform": [
              {
                "translateX": 56,
              },
            ],
            "writingDirection": "ltr",
          }
        }
        testID="animated-fab-text"
      />
    </View>
  </View>
</View>
`;

exports[`renders animated fab with label on the left 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "transparent",
      "borderRadius": 16,
      "opacity": 1,
      "position": "absolute",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="animated-fab-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 16,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    testID="animated-fab-container"
  >
    <View
      collapsable={false}
      style={
        [
          false,
          {
            "height": 56,
          },
          {
            "borderRadius": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "elevation": 0,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 3,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 72,
            }
          }
          testID="animated-fab-extended-shadow"
        />
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 3,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 56,
            }
          }
          testID="animated-fab-shadow"
        />
      </View>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          [
            {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 16,
              "height": 56,
              "left": -16,
              "right": undefined,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 72,
            }
          }
        >
          <View
            accessibilityLabel="text"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 16,
                },
              ]
            }
            testID="animated-fab"
          >
            <View
              style={
                [
                  {
                    "height": 56,
                  },
                  {
                    "borderRadius": 16,
                    "width": 72,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "height": 56,
          "justifyContent": "center",
          "left": -16,
          "position": "absolute",
          "right": undefined,
          "transform": [
            {
              "translateX": 16,
            },
          ],
          "width": 56,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          [
            {
              "color": "rgba(33, 0, 93, 1)",
              "fontSize": 24,
            },
            [
              {
                "lineHeight": 24,
                "transform": [
                  {
                    "scaleX": 1,
                  },
                ],
              },
              {
                "backgroundColor": "transparent",
              },
            ],
          ]
        }
      >
        plus
      </Text>
    </View>
    <View
      pointerEvents="none"
    >
      <Text
        collapsable={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        onTextLayout={[Function]}
        style={
          {
            "color": "rgba(33, 0, 93, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "left": 16,
            "letterSpacing": 0.1,
            "lineHeight": 20,
            "minWidth": 0,
            "opacity": 0,
            "position": "absolute",
            "textAlign": "left",
            "top": -28,
            "transform": [
              {
                "translateX": 56,
              },
            ],
            "writingDirection": "ltr",
          }
        }
        testID="animated-fab-text"
      >
        text
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders animated fab with label on the right by default 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "transparent",
      "borderRadius": 16,
      "opacity": 1,
      "position": "absolute",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 8,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
  testID="animated-fab-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 16,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      }
    }
    testID="animated-fab-container"
  >
    <View
      collapsable={false}
      style={
        [
          false,
          {
            "height": 56,
          },
          {
            "borderRadius": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "elevation": 0,
            },
          ]
        }
      >
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 3,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 72,
            }
          }
          testID="animated-fab-extended-shadow"
        />
        <View
          collapsable={false}
          pointerEvents="none"
          style={
            {
              "borderRadius": 16,
              "bottom": 0,
              "elevation": 3,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": [
                {
                  "translateX": 16,
                },
              ],
              "width": 56,
            }
          }
          testID="animated-fab-shadow"
        />
      </View>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          [
            {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            {
              "borderRadius": 16,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 16,
              "height": 56,
              "left": 16,
              "right": undefined,
              "transform": [
                {
                  "translateX": 0,
                },
              ],
              "width": 72,
            }
          }
        >
          <View
            accessibilityLabel="text"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 16,
                },
              ]
            }
            testID="animated-fab"
          >
            <View
              style={
                [
                  {
                    "height": 56,
                  },
                  {
                    "borderRadius": 16,
                    "width": 72,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      pointerEvents="none"
      style={
        {
          "alignItems": "center",
          "height": 56,
          "justifyContent": "center",
          "left": 16,
          "position": "absolute",
          "right": undefined,
          "transform": [
            {
              "translateX": 0,
            },
          ],
          "width": 56,
        }
      }
    >
      <Text
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          [
            {
              "color": "rgba(33, 0, 93, 1)",
              "fontSize": 24,
            },
            [
              {
                "lineHeight": 24,
                "transform": [
                  {
                    "scaleX": 1,
                  },
                ],
              },
              {
                "backgroundColor": "transparent",
              },
            ],
          ]
        }
      >
        plus
      </Text>
    </View>
    <View
      pointerEvents="none"
    >
      <Text
        collapsable={false}
        ellipsizeMode="tail"
        numberOfLines={1}
        onTextLayout={[Function]}
        style={
          {
            "color": "rgba(33, 0, 93, 1)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "500",
            "letterSpacing": 0.1,
            "lineHeight": 20,
            "minWidth": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 16,
            "textAlign": "left",
            "top": -28,
            "transform": [
              {
                "translateX": 56,
              },
            ],
            "writingDirection": "ltr",
          }
        }
        testID="animated-fab-text"
      >
        text
      </Text>
    </View>
  </View>
</View>
`;
